base/src/ImageViewerModule.cpp

```diff
@@ -77,24 +187,96 @@ bool ImageViewerModule::term() { return Module::term(); }
 
 bool ImageViewerModule::process(frame_container& frames)
 {
-	auto frame = getFrameByType(frames, FrameMetadata::RAW_IMAGE);
-	if (isFrameEmpty(frame))
+	mDetail->inputFrame = frames.cbegin()->second;
+	if (isFrameEmpty(mDetail->inputFrame))
 	{
 		return true;
 	}
-	
-	mDetail->showImage(frame);
+	mDetail->compute();
 	return true;
 }
 
 bool ImageViewerModule::processSOS(frame_sp& frame)
 {
-	auto metadata = frame->getMetadata();
-	mDetail->setMatImg(FrameMetadataFactory::downcast<RawImageMetadata>(metadata));
+	auto inputMetadata = frame->getMetadata();
+	auto frameType = inputMetadata->getFrameType();
+	FrameMetadata::MemType mInputMemType = inputMetadata->getMemType();
+    #if defined(__arm__) || defined(__aarch64__)
+	    int width = 0;
+        int height =0;
+		switch (frameType)
+        {
+        case FrameMetadata::FrameType::RAW_IMAGE:
+        {
+           auto metadata = FrameMetadataFactory::downcast<RawImageMetadata>(inputMetadata);
+           width = metadata->getWidth();
+           height = metadata->getHeight();
+        }
+        break;
+        case FrameMetadata::FrameType::RAW_IMAGE_PLANAR:
+        {
+           auto metadata = FrameMetadataFactory::downcast<RawImagePlanarMetadata>(inputMetadata);
+           width = metadata->getWidth(0);
+           height = metadata->getHeight(0);
+        }
+        break;
+        default:
+        throw AIPException(AIP_FATAL, "Unsupported FrameType<" + std::to_string(frameType) + ">");
+        }
+
+		mDetail->eglInitializer(height,width);
+	#else
+		mDetail->setMatImg(FrameMetadataFactory::downcast<RawImageMetadata>(inputMetadata));
+	#endif
 	return true;
 }
 
 bool ImageViewerModule::shouldTriggerSOS()
 {
 	return mDetail->shouldTriggerSOS();
+}
+
+bool ImageViewerModule::handleCommand(Command::CommandType type, frame_sp &frame)
+{
+#if defined(__arm__) || defined(__aarch64__)
+    if (type == Command::CommandType::DeleteWindow)
+    {
+        EglRendererCloseWindow cmd;
+        getCommand(cmd, frame);
```

Comment:
no point in deserializing this frame here

Reply:
yeah it is not required we are not passing anything to destroy window , removed it